Sample.java

```diff
@@ -1,7 +1,110 @@
-// Time Complexity :
-// Space Complexity :
-// Did this code successfully run on Leetcode :
-// Any problem you faced while coding this :
+# Hashing-2
+
+## Problem1 (https://leetcode.com/problems/subarray-sum-equals-k/)
+class Solution {
+    public int subarraySum(int[] nums, int k) {
+        if(nums==null || nums.length==0)
+        {
+            return 0;
+        }
+        int rsum=0,count=0;
+        HashMap<Integer, Integer> sMap=new HashMap<>();
+        sMap.put(0,1);
+        for(int i=0;i<nums.length;i++)
+        {
+            rsum=rsum+nums[i];
+            int s=rsum-k;
+                if(sMap.containsKey(s))
+                {
+                    count=count+sMap.get(s);
+                }
+            if(!sMap.containsKey(rsum))
+                sMap.put(rsum,1);
+            else
+                sMap.put(rsum,sMap.getOrDefault(rsum,0)+1);
+        }
+        return count;
+    }
+}
+// Time Complexity :o(n)
+// Space Complexity :o(1)
+// Did this code successfully run on Leetcode : YES 
+// Any problem you faced while coding this : NO 
+
```

Comment:
you should consider the hashmap that is being used for the space complexity for problem 1 and 2